esp8266/modnetwork.c

```diff
@@ -336,6 +336,15 @@ STATIC mp_obj_t esp_config(size_t n_args, const mp_obj_t *args, mp_map_t *kwargs
                         cfg.ap.channel = mp_obj_get_int(kwargs->table[i].value);
                         break;
                     }
+                    case QS(MP_QSTR_hostname): {
+                        req_if = STATION_IF;
+                        if (self->if_id == STATION_IF) {
+                            mp_uint_t len;
+                            const char *s = mp_obj_str_get_data(kwargs->table[i].value, &len);
```

Comment:
Thanks for the pointer to that.  I've switched to it and ran through some tests with it.